components/youtube/new-videos-in-playlist.js

```diff
@@ -0,0 +1,57 @@
+const axios = require("axios");
+const youtube = require("https://github.com/PipedreamHQ/pipedream/components/youtube/youtube.app.js");
+
+module.exports = {
+  name: "Youtube - New Videos in Playlist",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    youtube,
+    playlistId: {
+      type: "string",
+      label: "Playlist ID",
+      description: "The ID of the playlist to search for new videos in.",
+      default: "",
+    },
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  methods: {},
+
+  async run(event) {
+    let videos = [];
```

Comment:
Another thing I wanted to ask: are you planning to keep track of the last page token / video ID you pulled the last time the source ran? That is: when the source runs the first time it pulls the last N videos. Since you've already processed those N videos, the next time you can just pull videos starting at N + 1 moving forward.

You may have noticed that, by deduping on video ID in the call to `this.$emit`, if you pull duplicates, we won't emit them. But I was just curious if that was intentional, or if you planned to handle this in the API request to Youtube.

Reply:
The only issue there is that if someone deletes a video, then the page count would be off. There is a publishedAfter parameter I could include though to cut down on requests.

Reply:
Great, yeah have you seen our built-in `db` ([docs here](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#db))? I think you've used `$checkpoint` in workflows in the past - this is a similar API where you could store the video of the last ID you fetched in the prior run, etc.